src/System.IO.FileSystem/src/System/IO/FileStatus.Unix.cs

```diff
@@ -189,14 +193,18 @@ internal DateTimeOffset GetLastWriteTime(ReadOnlySpan<char> path, bool continueO
         }
 
         internal void SetLastWriteTime(string path, DateTimeOffset time)
-            => SetAccessWriteTimes(path, null, time.ToUnixTimeSeconds());
+        {
+            long secs = time.ToUnixTimeSeconds();
+            long milli = time.ToUnixTimeMilliseconds() - secs * 1000;
+            SetAccessWriteTimes(path, null, null, secs, milli * 1000000);
```

Comment:
Write numbers as eg 100_000 for readability?

Reply:
did you mean ```SetAccessWriteTimes(path, null, null, secs, milli * 1000 * 1000);``` ?

Reply:
you can write `1000000` as `1_000_000` since C# 6.